File: kinoafisha/Model/CinemasContainer.m
//
//  CinemasContainer.m
//  kinoafisha
//
//  Created by Eugene Dorfman on 11/9/14.
//  Copyright (c) 2014 justadreamer. All rights reserved.
//

#import "CinemasContainer.h"
#import "Cinema.h"

@implementation CinemasContainer
+ (NSDictionary *) JSONKeyPathsByPropertyKey {
    return @{
             @"cityName" : @"city_name",
             @"cinemas" : @"cinemas"
             };
}

+ (NSValueTransformer *) cinemasJSONTransformer {
    return [MTLJSONAdapter arrayTransformerWithModelClass:[Cinema class]];
}

@end

File: kinoafisha/CinemasContainer.swift
//
//  CinemasContainer.swift
//  kinoafisha
//
//  Created by Eugene Dorfman on 13/7/19.
//  Copyright © 2019 justadreamer. All rights reserved.
//

import Foundation

struct CinemasContainer: Codable {

    enum CodingKeys: String, CodingKey {
        case cityName = "city_name"
        case cinemas
    }

    var cityName: String
    var cinemas: [Cinema]
}

extension CinemasContainer: ProvidesEmptyState {
    static var empty: Self {
        return CinemasContainer(cityName: "", cinemas: [])
    }
}
